test/onnx/test_pytorch_onnx_shape_inference.py

```diff
@@ -205,8 +205,71 @@ def test_pad(self):
         constant = self.insert_tensor_constant(g, torch.ones(6, dtype=torch.long))
         none = g.op("prim::Constant").setType(torch.NoneType.get())
         pad = g.op("Pad", input, constant, none, mode_s="constant")
+        self.run_test(g, pad.node(), expect_tensor("Float", shape=(5, 322, 102)))
+
+    def test_pad_with_dynamic_input_shape(self):
+        g = self.create_empty_graph()
+        input = g.addInput()
+        input.setType(input.type().with_dtype(torch.float).with_sizes([3, None, None]))
+        constant = self.insert_tensor_constant(g, torch.ones(6, dtype=torch.long))
+        none = g.op("prim::Constant").setType(torch.NoneType.get())
+        pad = g.op("Pad", input, constant, none, mode_s="constant")
+        self.run_test(g, pad.node(), expect_tensor("Float", shape=(5, None, None)))
+
+    def test_pad_with_dynamic_pad_size(self):
```

Comment:
I would add the expected behavior in the test name. For example, in this case do we expect the shape is fully unknown? I would do the same for the previous test. Do we expect only the first dimension is known? I would also add a short comment before 217 to explain why the expected shape is such so that we know exactly what the test is expecting just by reading.

Reply:
I think it is repetitive to add in test name, since all behavior is to expect inferred type to match "whatever in `expect_tensor`". 
Most of the reasoning for behavior is pretty self-explanatory I feel, I'm not sure adding something like `# Only first dimension is static, because only first dimension in input is static.` is providing any extra information... If to expand the explanation then it would be much longer, but more like explaining the shape inference logic which is not exclusive to a certain test case.
I can add a short line explaining the expectation for resize with concat case, since the value folding propagation is not obvious.

Reply:
Discussed and resolved.